### PR TITLE
Unexpand: Initial implementation

### PR DESCRIPTION
**Pull request by bojan88**
Implementation for https://github.com/GrayJack/coreutils/issues/7

### CODE REVIEW

File: unexpand/src/tab_stops.rs [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+                if tabs_str == "" {
+                    return TabStops { offset: None, repetable: Some(8), positions: vec![] };
+                }
+
+                let tabs_str = tabs_str.replace(", ", ",");
+                let mut tabs_vec: Vec<&str> =
+                    tabs_str.split(|c| c == ',' || c == ' ').map(|s| s.trim()).collect();
+
+                if tabs_vec.len() == 1 {
+                    let value = tabs_vec[0].parse::<usize>().unwrap();
```
Comment (GrayJack) [inline unexpand/src/tab_stops.rs]: There is a awful lot of unwrapping in the code, there is any guarantee that these uses of unwrap won't panic?

File: unexpand/src/unexpand.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+name: unexpand
+version: "0.1"
+author: "Bojan Durdevic <bojan88@gmail.com>"
+about: Convert spaces to tabs
+args:
+    - all:
+        help: convert all blanks, instead of just initial blanks
```
Comment (GrayJack) [inline unexpand/src/unexpand.yml]: All help start with upper case

File: unexpand/src/unexpand.yml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+        long: first-only
+    - tabs:
+        help: have tabs N characters apart instead of 8 (enables -a)
+        short: t
+        long: tabs
+        value_names: N or LIST
+        takes_value: true
+    - version:
+        help: output version information and exit
+        long: version
```
Comment (GrayJack) [inline unexpand/src/unexpand.yml]: I think that is not needed (help and version are added by default on clap)

File: unexpand/src/unexpand.yml [hunk 4]
```diff
@@ -0,0 +1,24 @@
+name: unexpand
+version: "0.1"
+author: "Bojan Durdevic <bojan88@gmail.com>"
+about: Convert spaces to tabs
+args:
+    - all:
+        help: convert all blanks, instead of just initial blanks
+        short: a
+        long: all
+    - first_only:
```
Comment (GrayJack) [inline unexpand/src/unexpand.yml]: All options/flags should have a short and a long one, it it doesn't have on the original, you are free to create one

**Comment by bojan88** [comment]
@GrayJack thanks for the quick review. I appreciate it, especially the error handling parts.
I hope I addressed all your suggestions/requests.
Can you please review it again?

File: unexpand/src/tab_stops.rs [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+pub struct TabStops {
+    offset:    Option<usize>,
+    repetable: Option<usize>,
+    positions: Vec<usize>,
+}
+
+const ARG_PARSE_MSG: &str = "unexpand: error parsing arguments";
+
+impl TabStops {
+    pub fn new(tabs_str: Option<&str>) -> Result<TabStops, &str> {
```
Comment (bojan88) [inline unexpand/src/tab_stops.rs]: I'm not sure, but I have a feeling that `new` should always return the structure instance.
Maybe I should rename this method...

Comment (GrayJack) [reply unexpand/src/tab_stops.rs]: Nah, that is ok in cases where the struct cannot be safely constructed  without check, as example, checkout `CString::new()` on `std`

**Comment by GrayJack** [comment]
Looks fine for me :smile: 

The test grow big, so I would recommend create a module just for tests, give a look at the `wc` and `cut`, that's the way I did it the tests module

**Comment by bojan88** [comment]
@GrayJack I moved the tests to a separate module, but in the same file. Hope that's fine...

**Comment by GrayJack** [comment]
It's great!! Thanks for the contribution

**Pull request merged**